Library/Homebrew/diagnostic.rb

```diff
@@ -1435,5 +1420,34 @@ def all
         methods.map(&:to_s).grep(/^check_/)
       end
     end # end class Checks
+
+  end
+
+  def inject_dump_stats!(the_module, pattern = /.*/)
+    the_module.module_eval do
+      instance_methods.grep(pattern).each do |name|
+        method = instance_method(name)
+        define_method(name) do |*args, &block|
+          begin
+            time = Time.now
+            method.bind(self).call(*args, &block)
+          ensure
+            $times[name] ||= 0
+            $times[name] += Time.now - time
+          end
+        end
+      end
+    end
+
+    if $times.nil?
```

Comment:
Not right now, but why write code to be brittle like that when being robust is easy?

I added a check to make sure that methods are not double-instrumented, either.